lib/auth/auth_repository.dart

```diff
@@ -13,33 +14,43 @@ mixin AuthRepositoryLoggy implements LoggyType {
 }
 
 class AuthRepository with AuthRepositoryLoggy {
-  static AuthRepository get instance => GetIt.instance<AuthRepository>();
-
   AuthRepository({required AuthApi authApi}) : _authApi = authApi;
 
+  final _authStreamController = BehaviorSubject<AuthenticationStatus>();
+
   final AuthApi _authApi;
-  late final Fresh<OAuth2Token> _fresh = Fresh.oAuth2(
-    tokenStorage: InMemoryTokenStorage<OAuth2Token>(),
-    refreshToken: (token, client) async {
-      loggy.debug('refreshing token...');
-      final refreshedToken = await _authApi.refreshToken(token);
-      loggy.debug('token refreshed!');
-      return refreshedToken;
-    },
-    shouldRefresh: (Response? response) => response?.statusCode == 401,
-  );
-
-  Stream<AuthenticationStatus> get authStatus => _fresh.authenticationStatus;
+  late StreamSubscription freshSubscription;
+  late Fresh<OAuth2Token> freshInstance;
+
+  Stream<AuthenticationStatus> get authStatus =>
+      _authStreamController.asBroadcastStream();
+
+  AuthenticationStatus get currentAuthStatus => _authStreamController.value;
```

Comment:
Why creating a controller/stream that emits when the built-in fresh stream emits?

Because of this `currentAuthStatus` property. We can't easily read the current state of the authentication with the `fresh.authenticationStatus` stream.

`fresh.authenticationStatus.last` is async. It returns a Future that is resolved with the latest value only when the stream is closed (and it never is)